### PR TITLE
Lose allocation when retiring or hw failing

### PR DESCRIPTION
**Pull request by hakonhall**

### CODE REVIEW

**Comment by hakonhall** [comment]
FYI @freva

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java [hunk 1]
```diff
@@ -470,8 +470,8 @@ public Node fail(String hostname, Agent agent, String reason) {
      * @return the node in its new state
      * @throws NoSuchNodeException if the node is not found
      */
-    public Node park(String hostname, Agent agent, String reason) {
-        return move(hostname, Node.State.parked, agent, Optional.of(reason));
+    public Node park(String hostname, boolean keepAllocation, Agent agent, String reason) {
```
Comment (freva) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java]: Keep the same order of arguments as `move()`

**Review by mpolden — APPROVED** [review]
💯

File: node-repository/src/test/java/com/yahoo/vespa/hosted/provision/maintenance/NodeRetirerTester.java [hunk 2]
```diff
@@ -163,9 +164,9 @@ void assertCountsForStateByFlavor(Node.State state, long... nums) {
     }
 
     void assertParkedCountsByApplication(long... nums) {
-        Map<ApplicationId, Long> expected = expectedCountsByApplication(nums);
-        Map<ApplicationId, Long> actual = nodeRepository.getNodes(Node.State.parked).stream()
-                .collect(Collectors.groupingBy(node -> node.allocation().get().owner(), Collectors.counting()));
+        // Nodes loose allocation when parked, so just do a sum.
```
Comment (mpolden) [inline node-repository/src/test/java/com/yahoo/vespa/hosted/provision/maintenance/NodeRetirerTester.java]: loose -> lose

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java (hunk 1 above)
Comment (hakonhall) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java]: Done

File: node-repository/src/test/java/com/yahoo/vespa/hosted/provision/maintenance/NodeRetirerTester.java (hunk 2 above)
Comment (hakonhall) [reply node-repository/src/test/java/com/yahoo/vespa/hosted/provision/maintenance/NodeRetirerTester.java]: Done

**Review by freva — APPROVED** [review]

**Pull request merged**